Lib/configparser.py

```diff
@@ -724,16 +724,25 @@ def read(self, filenames, encoding=None):
         filename may also be given.
 
         Return list of successfully read files.
+
+        Any exception other than an OSError raised while
+        parsing the file is propogated to the caller.
         """
         if isinstance(filenames, (str, bytes, os.PathLike)):
             filenames = [filenames]
         encoding = io.text_encoding(encoding)
         read_ok = []
         for filename in filenames:
             try:
-                with open(filename, encoding=encoding) as fp:
+                fp = open(filename, encoding=encoding)
+            except (OSError, ValueError):
+                continue
+            try:
+                with fp:
                     self._read(fp, filename)
             except OSError:
```

Comment:
I wonder if it's still necessary to trap `OSError` while reading. I removed this try/except and all tests still pass. Maybe we should just assume that `OSError` is there to handle the `open` call and remove it from here (and also remove the change to the docstring).

Reply:
That's what I thought in the first place but since it was already there beforehand, I wondered whether we needed to trap it or not. Personally, I would have preferred not to but I wanted to retain a possible backward compatibility (though I don't know if it's common to override the parser =/)

Reply:
This is documented. See "Files that cannot be opened are silently ignored" in the docstring.

If tests are passed, this is a lack of testing.

Reply:
Yes but it also says "Return list of successfully read files". And for me, opening and reading are distinct. The first is only for the open() call, the second considers the parsing phase (which includes decoding the content).